src/Connect.js

```diff
@@ -337,59 +332,123 @@ class Connect {
+   * effectively logging them out. The keypair (app-instance identity) is preserved
    */
-  getState() {
-    const connectState = store.get('connectState')
-    if (connectState) this.deserialize(connectState)
+  logout() {
+    // Clear explicit state
+    this.setState({
+      did: null, 
+      mnid: null, 
+      address: null, 
+      doc: null, 
+      pushToken: null, 
+      publicEncKey: null
+    })
+
+    // Clear all instance variables with references to current state
+    this.pushTransport = null
+  }
+
+  /**
+   * Clear the entire state of the connect instance, including the keypair, from memory
+   * and localStorage.  Rebuild this.credentials with a new app-instance identity
+   */
+  reset() {
+    this.logout()
+    // Rebuild credentials
+    this.keypair = Credentials.createIdentity()
+    this.credentials = new Credentials({...this.keypair, ...this.resolverConfigs})
   }
 
-    /**
-     *  Writes serialized uPort connect state to browser localStorage at key 'connectState'
-     */
-  setState() {
-    const connectState = this.serialize()
-    store.set('connectState', connectState)
+  /**
+   * Accessor methods for Connect state.  The state consists of the key-value pairs below
+   *  (did, doc, mnid, address, keypair, pushToken, and publicEncKey)
+   */
+  get did ()          { return this._state.did }
+  get doc ()          { return {...this._state.doc} }
+  get mnid ()         { return this._state.mnid }
+  get address ()      { return this._state.address }
+  get keypair ()      { return {...this._state.keypair} }
+  get pushToken ()    { return this._state.pushToken }
+  get publicEncKey () { return this._state.publicEncKey }
+
+  /**
+   * Setter methods with appropriate validation
+   */
+  set did (did)                   { this.setState({did}) }
+  set doc (doc)                   { this.setState({doc}) }
+  set mnid (mnid)                 { this.setState(this.mnidDecode(mnid)) }
+  set address (address)           { this.setState(this.mnidDecode(address)) }
+  set keypair (keypair)           { this.setState({keypair}) }
+  set pushToken (pushToken)       { this.setState({pushToken}) }
+  set publicEncKey (publicEncKey) { this.setState({publicEncKey}) }
+
+  /**
+   * Utility method for disambiguating an address or mnid; 
+   * Receives either and return an object containing both, encoded according to this.network.id
+   *
+   * @param   {String}  addressOrMnid   -- A string containing an address or an mnid
+   * @returns {Object}  addressAndMnid  -- an object with propreties address, and mnid containing both
+   */
+  mnidDecode(addressOrMnid) {
+    if (!addressOrMnid) return {adddress: undefined, mnid: undefined}
+    const address = isMNID(addressOrMnid) ? decode(addressOrMnid).address : addressOrMnid
+    const mnid = isMNID(addressOrMnid) ? addressOrMnid : encode({network: this.network.id, address: addressOrMnid})
```

Comment:
hmm, I think we will want different constraints in this function, a given address does not necessarily come from this.network.id, if not given an mnid, information (net id) is lost that we can not recover. What would probably makes sense, is:

if given only mnid -> decode address and return both address, mnid
if given only address -> throw error (unless matches current mnid, no change)

then above only set mnid (not both address, and mnid)